.github/workflows/build.yml

```diff
@@ -19,10 +22,30 @@ jobs:
       matrix:
         mode: ["", tiny, rel, tinylinux, optlinux]
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+        with:
+          # Full checkout needed for git-restore-mtime-bare.
+          fetch-depth: 0
+
+      # TODO(jart): fork this action.
+      - uses: chetan/git-restore-mtime-action@v2
+
+      - uses: actions/cache@v4
+        with:
+          path: .cosmocc/${{ env.COSMOCC_VERSION }}
+          key: cosmocc-${{ env.COSMOCC_VERSION }}
+
+      - uses: actions/cache@v4
+        with:
+          path: o
```

Comment:
It might be fun to see if `ln -sfn $(mktemp -d /dev/shm/o.XXXXXXXX) o` speeds up the linux builds further.